### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by LovelyDrug**
[DEMO LINK](https://LovelyDrug.github.io/react_dynamic-list-of-todos/)

### CODE REVIEW

**Review by blcts — APPROVED** [review]
nice

File: src/App.tsx [hunk 1]
```diff
@@ -1,14 +1,73 @@
...
+    setIsLoading(true);
+
+    getTodos()
+      .then((todo) => {
+        setTodos(todo);
+        setIsLoading(false);
+      });
+  }, []);
+
+  const getFilter = (option: string) => {
```
Comment (VETALEK) [inline src/App.tsx:55]: missed `useCallback` here

File: src/components/TodoModal/TodoModal.tsx [hunk 2]
```diff
@@ -1,12 +1,38 @@
...
+  selectedTodoId: number;
+  onSelect: (id: number | null) => void;
+}
+
+export const TodoModal: React.FC<Props> = (props) => {
+  const { selectedTodoId, todos, onSelect } = props;
+  const [isModalLoading, setIsModalLoading] = useState(false);
+  const [user, setUser] = useState<User | null>(null);
+
+  const getSelectedTodo = () => {
```
Comment (VETALEK) [inline src/components/TodoModal/TodoModal.tsx:18]: `useCallback`

**Review by VETALEK — CHANGES_REQUESTED** [review]
GJ 👍 
commented

**Review by VETALEK — APPROVED** [review]
🔥

**Pull request closed without merging**